contracts/multisigs/GuardCM.sol

```diff
@@ -0,0 +1,292 @@
+        if (proposalId == 0) {
+            revert ZeroValue();
+        }
+
+        governorCheckProposalId = proposalId;
+        emit GovernorCheckProposalIdChanged(proposalId);
+    }
+
+    /// @dev Verifies authorized target and selector in the schedule or scheduleBatch function call.
+    /// @param data Data in bytes.
+    /// @param selector Schedule function selector.
+    function _verifySchedule(bytes memory data, bytes4 selector) internal view {
+        //address,uint256,bytes,bytes32,bytes32,uint256
+        bytes memory payload = new bytes(data.length - 4);
+        for (uint256 i = 0; i < payload.length; ++i) {
+            payload[i] = data[i + 4];
+        }
+
+        // Prepare the decoding data sets
+        address[] memory targets;
+        bytes[] memory callDatas;
+        if (selector == SCHEDULE) {
+            targets = new address[](1);
+            callDatas = new bytes[](1);
+            // Decode the data in the schedule function
+            (targets[0], , callDatas[0], , , ) =
+                abi.decode(payload, (address, uint256, bytes, bytes32, bytes32, uint256));
+        } else {
+            // Decode the data in the scheduleBatch function
+            (targets, , callDatas, , , ) =
+            abi.decode(payload, (address[], uint256[], bytes[], bytes32, bytes32, uint256));
+        }
+
+        // Traverse all the schedule targets and selectors extracted from calldatas
+        for (uint i = 0; i < targets.length; ++i) {
+            // Push a pair of key defining variables into one key
+            // target occupies first 160 bits
+            uint256 targetSelector = uint256(uint160(targets[i]));
+            // selector occupies next 32 bits
+            targetSelector |= uint256(uint32(bytes4(callDatas[i]))) << 160;
+
+            // Check the authorized combination of target and selector
+            if (!mapAllowedTargetSelectors[targetSelector]) {
+                revert NotAuthorized(targets[i], bytes4(callDatas[i]));
+            }
+        }
+    }
+
+    /// @dev Checks the transaction for authorized arguments.
+    /// @notice Scheduling in timelock in checked against authorized targets and signatures.
+    /// @notice No self-multisig function calls are allowed.
+    /// @param to Destination address of Safe transaction.
+    /// @param data Data payload of Safe transaction.
+    /// @param operation Operation type of Safe transaction.
+    function checkTransaction(
+        address to,
+        uint256,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256,
+        uint256,
+        uint256,
+        address,
+        address payable,
+        bytes memory,
+        address
+    ) external view {
+        // Just return if paused
+        if (paused == 1) {
+            // Call to the timelock
+            if (to == owner) {
+                // No delegatecall is allowed
+                if (operation == Enum.Operation.DelegateCall) {
```

Comment:
why not delegate call or seld-multisig call?

Reply:
- delegatecall to timelock does not make practical sense, but we do not need storage collisions when timelock can be used as a library to change storage of Guard.
- self-multisig not allowed for change owners, thershould, etc

Reply:
Because we don't want the multisig to change itself. And the delegatecall... is super unexpected here when calling the Timelock. This means that the multisig wants to joint something in the storage with the Timelock - super weird ).